mantidproject/mantid
# Calculation of crystal electric field energies

Translated a fortran subroutine into C++. Exposed the new function into python. There is no direct way of exporting a function form a Mantid plugin library. As a workaround an algorithm is written and used by a python script. The algorithm is not intended for any other use so no documentation page was created.

**To test:**

This script should be able to run

```
from CrystalField.energies import energies
en, wf, ham = energies(1,  B20=0.3365, B22=7.4851, B40=0.4062, B42=-3.8296, B44=-2.3210)
```

without crashing and produce a vector (en) and 2 complex non-zero matrices (wf and ham). 

Fixes #15207.

It is a part of a larger issue #15684 and it's too early for the release notes. Added an entry where they will go.

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
